languages/ramldoc/src/main/kotlin/io/vrap/codegen/languages/ramldoc/extensions/VrapExtensions.kt

```diff
@@ -67,23 +95,136 @@ fun UnionType.renderUnionType(): String {
+    if (this.items != null) {
+        t += """
+                |items:
+                |  <<${this.items.renderTypeFacet()}>>
+            """
+    }
+    return t.trimMargin().keepIndentation("<<", ">>")
+}
+
+fun UnionAnnotationType.renderUnionType(): String {
+    val typeString = this.oneOf.joinToString(" | ") { when(it) { is ArrayType -> "${it.items.name}[]" else -> it.name } }
+
+    val unionString = """
+        |(oneOf):
+        |${this.oneOf.joinToString("\n") { when(it) { is ArrayType -> "- ${it.items.name}[]" else -> "- ${it.name}" } }}
+    """.trimMargin()
+
+    val t = listOf<String>().plus("type: ${typeString}").plus(unionString).plus(this.renderEAttributes()).joinToString("\n")
+    return t.trimMargin().keepIndentation("<<", ">>")
+}
+
+fun AnyAnnotationType.renderTypeFacet(): String {
+    return when (this) {
+        is ArrayAnnotationType -> this.renderArrayType()
+        is UnionAnnotationType -> this.renderUnionType()
+        is ObjectAnnotationType -> this.renderObjectType()
+        else -> this.renderScalarType()}
+}
+
+fun AnyAnnotationType.renderType(withDescription: Boolean = true): String {
     val builtinType = "(builtinType): ${BuiltinType.of(this.eClass()).map { it.getName() }.orElse("any")}"
     val description = if (withDescription && this.description?.value.isNullOrBlank().not()) {
         """
         |description: |-
         |  <<${this.description.value.trim()}>>
-        """.trimEnd().keepIndentation("<<", ">>")
+        """.trimMargin().keepIndentation("<<", ">>")
     } else {
         ""
     }
     return """
-        |${when (this) {
-            is ArrayType -> this.renderArrayType()
-            is UnionType -> this.renderUnionType()
-            else -> this.renderScalarType()}}
-        |$builtinType$description
-        """.trimMargin()
+        |${this.renderTypeFacet()}
+        |$builtinType
+        |$description
+        """.trimMargin().trimEnd()
+}
+
+fun VrapEnumType.packageDir(prefix: String): String {
+    val dir = this.`package`.replace(prefix, "").trim('/')
+    return dir + if (dir.isNotEmpty()) "/" else ""
+}
+
+fun VrapObjectType.packageDir(prefix: String): String {
+    val dir = this.`package`.replace(prefix, "").trim('/')
+    return dir + if (dir.isNotEmpty()) "/" else ""
+}
+
+fun Annotation.renderAnnotation(): String {
+    return when (this.type) {
+        is ObjectAnnotationType -> """
+            |(${this.type.name}):
+            |  <<${this.value.toYaml()}>>""".trimMargin().keepIndentation("<<", ">>")
+        is ArrayAnnotationType -> """
+            |(${this.type.name}):
+            |  <<${this.value.toYaml()}>>""".trimMargin().keepIndentation("<<", ">>")
+        is StringAnnotationType ->
+            when(this.value) {
+                is ObjectInstance -> """
+                    |(${this.type.name}): |-
+                    |  <<${this.value.toJson()}>>""".trimMargin().keepIndentation("<<", ">>")
```

Comment:
Maybe you can dd an extension and call it `keepPhpIndentation` for example 🤔

Reply:
you saw we are not talking about PHP here, but RAML? 😄

Reply:
Question is more likely if we should change all keepIndentation methods to default to double brackets

Reply:
to be fair i feel like brakets in general cause problem 🤦‍♂ , maybe `{{}}` double accolades like handlebar would be a solution